domino_data/featurestore_sync.py

```diff
@@ -0,0 +1,213 @@
+                tags=FeatureViewRequestTags.from_dict(fv.tags),
+            )
+            request_input.append(feature_v)
+
+        # TODO update the API to include the commit id so that feature views
+        # and commit id can be updated to domino in one call
+        self.client.post_feature_views(request_input)
+        print("Feature Views successfully synced.")
+
+    def lock(self) -> None:
+        print("Locking the feature store")
+        # TODO Call feature store client lock method
+        # Raise FeatureStoreLockError with detailed reason if fails to lock.
+        print("Locked the feature store")
+        return True
+
+    def unlock(self) -> None:
+        print("UnLocking the feature store")
+        # TODO Call feature store client unlock method
+        print("UnLocked the feature store")
+        return True
+
+
+def find_feast_repo_path():
+    sub_dirs = []
+    if not os.path.exists(FEAST_REPO_ROOT):
+        raise FileNotFoundError(f"The repo root path {FEAST_REPO_ROOT} does not exist.")
+    if not os.path.isdir(FEAST_REPO_ROOT):
+        raise NotADirectoryError(f"The repo root path {FEAST_REPO_ROOT} is not a directory.")
+
+    for file_name in os.listdir(FEAST_REPO_ROOT):
+        if file_name.startswith("."):
+            continue
+        full_path = os.path.join(FEAST_REPO_ROOT, file_name)
+        if os.path.isdir(full_path):
+            sub_dirs.append(full_path)
+
+    if len(sub_dirs) == 0:
+        raise FileNotFoundError(f"No repo is found under {FEAST_REPO_ROOT}")
+
+    if len(sub_dirs) > 1:
+        raise MultiFeastReposError(
+            f"Multiple repos found under {FEAST_REPO_ROOT}: " + ",".join(sub_dirs)
+        )
+
+    return sub_dirs[0]
+
+
+def pull_repo(repo):
+    logger.info(f"Commit Sha before pulling:{repo.head.object.hexsha}")
+    print("Pulling the repo......")
+    pull_result = repo.remotes.origin.pull()
+    result_flag = pull_result[0].flags
+    if result_flag == FetchInfo.ERROR or result_flag == FetchInfo.REJECTED:
+        raise GitPullError(f"Failed to pull the repo with error flag {result_flag}")
+    print("Finished pulling the repo.")
+    print(f"Commit Sha after pulling:{repo.head.object.hexsha}")
+
+
+def run_command(args, cwd):
+    """
+    Run the command with given arguments from given directory in a new process
+    :param args: a list representation of the command to run. e.g. ['feast', 'apply']
+    :param cwd: The directory from where to run the command
+    :return: a tuple of run status code and the output message. 0 means a successful run, other
+    status codes indicate failures
+    """
+    try:
+        return (
+            0,
+            subprocess.check_output(
+                args,
+                cwd=cwd,
+                stderr=subprocess.STDOUT,
+            ),
+        )
+    except subprocess.CalledProcessError as e:
+
+        return e.returncode, e.output
+
+
+def run_feast_apply():
```

Comment:
feast apply should be calling a python API, do we need to run a subcommand here?